src/entity/object/ItemEntity.php

```diff
@@ -226,12 +226,18 @@ public function onCollideWithPlayer(Player $player) : void{
 		}
 
 		$item = $this->getItem();
-		$playerInventory = $player->getInventory();
+		$playerInventory = match(true){
+			$player->getOffHandInventory()->canAddItem($item) and !$player->getOffHandInventory()->isSlotEmpty(0) => $player->getOffHandInventory(),
+			$player->getInventory()->canAddItem($item) => $player->getInventory(),
+			default => null
+		};
 
-		if($player->hasFiniteResources() and !$playerInventory->canAddItem($item)){
+		if($player->hasFiniteResources() and $playerInventory === null){
 			return;
 		}
 
+		$playerInventory ??= $player->getInventory();
```

Comment:
Tangential to this PR, but it is kinda misleading to provide an inventory for pickup that won't actually accept the item (even if it is creative). I guess that points to a needed API change (inventories don't actually pick up items anyway, their holders (blocks or entities) do).